src/algebra/module/pointwise_pi.lean

```diff
@@ -0,0 +1,50 @@
+/-
+Copyright (c) 2021 Alex J. Best. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex J. Best
+-/
+import algebra.pointwise
+import algebra.module.pi
+
+/-!
+# Pointwise actions on sets in Pi types
+
+This file contains lemmas about pointwise actions on sets in Pi types.
+
+## Tags
+
+set multiplication, set addition, pointwise addition, pointwise multiplication, pi
+
+-/
+
+open_locale pointwise
+open set
+
+variables {K ι : Type*} {R : ι → Type*}
+
+@[to_additive]
+lemma smul_pi_subset [∀ i, has_scalar K (R i)] (r : K) (t : Π i, set (R i)) (s : set ι) :
+  r • pi s t ⊆ pi s (r • t) :=
+begin
+  rintros x ⟨y, h, rfl⟩ i hi,
+  exact smul_mem_smul_set (h i hi),
+end
+
+@[to_additive]
+lemma smul_univ_pi [∀ i, has_scalar K (R i)] (r : K) (t : Π i, set (R i)) (s : set ι) :
+  r • pi (univ : set ι) t = pi (univ : set ι) (r • t) :=
+subset.antisymm (smul_pi_subset _ _ _) $ λ x h, begin
+  refine ⟨λ i, classical.some (h i $ set.mem_univ _), λ i hi, _, funext $ λ i, _⟩,
+  { exact (classical.some_spec (h i _)).left, },
+  { exact (classical.some_spec (h i _)).right, },
+end
+
+@[to_additive]
+lemma smul_pi [group K] [∀ i, mul_action K (R i)] (r : K) (t : Π i, set (R i)) (S : set ι) :
+  r • S.pi t = S.pi (λ (i : ι), r • t i) :=
```

Comment:
To match the other lemmas, this should either be
```suggestion
  r • S.pi t = S.pi (r • t) :=
```
Or the other lemmas should match this.

I'm in favor of dropping the `λ`s and using `pi.has_scalar` as in that suggestion, but I'll let another reviewer make the call. No need to merge this suggestion until we've decided which spelling is better.

Reply:
Well we may aswell be consistent in the interim, I didn't get around to double checking what this broke in my proofs yet